Compile warnings could be eliminated

### Describe the Code Quality Issue

Warnings are generated during compile time:

```cpp
abacus-develop/source/module_io/restart.cpp:29:9: warning: add explicit braces to avoid dangling else [-Wdangling-else]
   29 |         else return false;
      |         ^
abacus-develop/source/module_io/restart.cpp:34:9: warning: add explicit braces to avoid dangling else [-Wdangling-else]
   34 |         else return false;
      |         ^

// where there is a risk of dangling else
if (-1 == file)
        if (error_quit)
            throw std::runtime_error("can't open restart save file. \nerrno=" + ModuleBase::GlobalFunc::TO_STRING(errno) + ".\n" + ModuleBase::GlobalFunc::TO_STRING(__FILE__) + " line " + ModuleBase::GlobalFunc::TO_STRING(__LINE__));
        else return false;
```
another warning
```cpp
abacus-develop/source/module_io/test_serial/read_input_test.cpp:28:1: warning: non-void function does not return a value [-Wreturn-type]
   28 | }
      | ^
```

Shall I modify this source file `restart.cpp` to eliminate these warnings?

### Additional Context

_No response_

### Task list for Issue attackers (only for developers)

- [ ] Identify the specific code file or section with the code quality issue.
- [ ] Investigate the issue and determine the root cause.
- [ ] Research best practices and potential solutions for the identified issue.
- [ ] Refactor the code to improve code quality, following the suggested solution.
- [ ] Ensure the refactored code adheres to the project's coding standards.
- [ ] Test the refactored code to ensure it functions as expected.
- [ ] Update any relevant documentation, if necessary.
- [ ] Submit a pull request with the refactored code and a description of the changes made.

--- a/source/module_io/restart.cpp
+++ b/source/module_io/restart.cpp
@@ -23,15 +23,21 @@ void Restart::read_file1(const std::string &file_name, void*const ptr, const siz
 bool Restart::write_file2(const std::string& file_name, const void* const ptr, const size_t size, const bool error_quit) const
 {
 	const int file = open(file_name.c_str(), O_WRONLY|O_CREAT|O_TRUNC, S_IRUSR|S_IWUSR);
-    if (-1 == file)
-        if (error_quit)
+    if (-1 == file){
+        if (error_quit){
             throw std::runtime_error("can't open restart save file. \nerrno=" + ModuleBase::GlobalFunc::TO_STRING(errno) + ".\n" + ModuleBase::GlobalFunc::TO_STRING(__FILE__) + " line " + ModuleBase::GlobalFunc::TO_STRING(__LINE__));
-        else return false;
+        } else {
+            return false;
+        }
+    }
     auto error = write(file, ptr, size);
-    if (-1 == error)
-        if (error_quit)
+    if (-1 == error) {
+        if (error_quit) {
             throw std::runtime_error("can't write restart save file. \nerrno=" + ModuleBase::GlobalFunc::TO_STRING(errno) + ".\n" + ModuleBase::GlobalFunc::TO_STRING(__FILE__) + " line " + ModuleBase::GlobalFunc::TO_STRING(__LINE__));
-        else return false;
+        } else {
+            return false;
+        }
+    }
     close(file);
     return true;
 }
@@ -39,15 +45,21 @@ bool Restart::write_file2(const std::string& file_name, const void* const ptr, c
 bool Restart::read_file2(const std::string& file_name, void* const ptr, const size_t size, const bool error_quit) const
 {
 	const int file = open(file_name.c_str(), O_RDONLY);
-    if (-1 == file)
-        if (error_quit)
+    if (-1 == file) {
+        if (error_quit) {
             throw std::runtime_error("can't open restart load file. \nerrno=" + ModuleBase::GlobalFunc::TO_STRING(errno) + ".\n" + ModuleBase::GlobalFunc::TO_STRING(__FILE__) + " line " + ModuleBase::GlobalFunc::TO_STRING(__LINE__));
-        else return false;
+        } else {
+            return false;
+        }
+    }
     auto error = read(file, ptr, size);
-    if (-1 == error)
-        if (error_quit)
+    if (-1 == error) {
+        if (error_quit) {
             throw std::runtime_error("can't read restart load file. \nerrno=" + ModuleBase::GlobalFunc::TO_STRING(errno) + ".\n" + ModuleBase::GlobalFunc::TO_STRING(__FILE__) + " line " + ModuleBase::GlobalFunc::TO_STRING(__LINE__));
-        else return false;
+        } else {
+            return false;
+        }
+    }
     close(file);
     return true;
 }

